From 04d3bb4574ac9ed926e8d73f93c56b00b7b2260d Mon Sep 17 00:00:00 2001
From: Andrew Zaborowski <andrew.zaborowski@intel.com>
Date: Wed, 9 Apr 2014 11:55:45 -0700
Subject: [PATCH] pstore: Decrease default max pstore dump sizes.

Re-re-apply 254958899d6847e785ecc0642065621eef926490 from kernel-uefi
aka. 102663f37fae1cd192e7c4b835f7bb01fdfa34a4 from kernel-starpeak.
Note in kernel-starpeak the concept of extra dumpers didn't exist but
the patches have now reappeared from somewhere so we bring back the
changes from the old kernel-uefi.

To save more space in pstore only allow about 5kB to be dumped per
crash, 4kB of the kmsg buffer containing interesting things like the
call trace and 1kB for "extra dumpers" -- there's only one enabled in
our kernel at this time, 1kB should give us the few last lines of logcat
if useful for telemetry.

Change-Id: I61c86d8c578e4dc41c31dd0ed3c80134dfffc179
Signed-off-by: Andrew Zaborowski <andrew.zaborowski@intel.com>
---
 drivers/staging/android/logger.c | 2 +-
 fs/pstore/platform.c             | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/android/logger.c b/drivers/staging/android/logger.c
index 1abcf55..90790e2 100644
--- a/drivers/staging/android/logger.c
+++ b/drivers/staging/android/logger.c
@@ -77,7 +77,7 @@ struct logger_log {
  * pstore is cleaned.
  */
 
-static ulong logger_pstore_size = 1500;
+static ulong logger_pstore_size = 1024;
 module_param_named(pstore_size, logger_pstore_size, ulong, 0600);
 MODULE_PARM_DESC(pstore_size,
 		"maximum bytes of pstore dump for each log buffer");
diff --git a/fs/pstore/platform.c b/fs/pstore/platform.c
index 2269d41..e86a6bb 100644
--- a/fs/pstore/platform.c
+++ b/fs/pstore/platform.c
@@ -49,7 +49,7 @@ MODULE_PARM_DESC(update_ms, "milliseconds before pstore updates its content "
 		 "enabling this option is not safe, it may lead to further "
 		 "corruption on Oopses)");
 
-static ulong pstore_extra_size = 8000;
+static ulong pstore_extra_size = 1024;
 module_param_named(extra_size, pstore_extra_size, ulong, 0600);
 MODULE_PARM_DESC(extra_size, "maximum of dumped extra data (beside kmsg)");
 
@@ -84,7 +84,7 @@ static char *big_oops_buf;
 static size_t big_oops_buf_sz;
 
 /* How much of the console log to snapshot */
-static unsigned long kmsg_bytes = 10240;
+static unsigned long kmsg_bytes = 4096;
 
 void pstore_set_kmsg_bytes(int bytes)
 {
-- 
1.8.3.2

